src/routes/give-away/create.js

```diff
@@ -0,0 +1,51 @@
+import models from '../../models'
+
+export default {
+
+  method: 'post',
+  path: '/giveaways?',
+
+  requireAuthentication: true,
+
+  handler(request, response) {
+
+    // @TODO: This is getting claned up by the orphan job
+    return models.CodexRecordFile.findById('5b1ef57078d795dc2bae3560')
+      .lean()
+      .then((leanMainImage) => {
+        delete leanMainImage._id
+
+        return models.CodexRecordFile.create(leanMainImage)
```

Comment:
Also, there's not a hard requirement that the `CodexRecordFile` id's be unique for each Giveaway, no? I would just delete this whole "copy" block and do:

```javascript
return models.CodexRecordFile.findById('5b1ef57078d795dc2bae3560')
  .then((giveawayMainImage) => {
```

So each Giveaway is using the same file record. Unfortunately we can't extend this concept to the Metadata record otherwise anyone who updated the metadata would be updating it for everyone else lol

Reply:
Unless it was caused by something else, pretty sure these will get cleaned up if one of metadata these files gets attached to is orphaned--it will delete the file for all other metadata that relies on it